src/Vector.cpp

```diff
@@ -0,0 +1,119 @@
+
+//
+// Created by Dvir Avraham on 19/03/2023.
+//
+
+#include "Vector.h"
+
+// implemnt vector
+
+// constructor with size and value
+Vector::Vector(size_t size, int value) {
+    _data = new int[size];
+    _size = size;
+    _capacity = size;
+    for (int i = 0; i < size; i++) {
+        _data[i] = value;
+    }
+}
+
+// destructor
+Vector::~Vector() {
+    delete[] _data;
+}
+
+//
+
+Vector::Vector(const Vector &other) {
+    _data = new int[other._size];
+    _size = other._size;
+    _capacity = other._size;
+    for (int i = 0; i < _size; i++) {
+        _data[i] = other._data[i];
+    }
+}
+
+Vector &Vector::operator=(const Vector &other) {
+    if (this != &other) {
+        delete[] _data;
+        _data = new int[other._size];
+        _size = other._size;
+        _capacity = other._size;
```

Comment:
The function sets _capacity equal to other._size, which could result in the vector needing to reallocate memory if the original vector was larger than other._size. This can lead to unnecessary memory allocations and deallocations, which can slow down the program.